### PR TITLE
Upgrade TypeScript to v5

### PR DESCRIPTION
**Pull request by johnnyomair**

### CODE REVIEW

File: admin/package.json [hunk 1]
```diff
@@ -89,7 +89,7 @@
         "npm-run-all": "^4.1.5",
         "prettier": "^2.2.1",
         "ts-node": "^10.0.0",
-        "typescript": "^4.2.3",
+        "typescript": "~5.1.0",
```
Comment (johnnyomair) [inline admin/package.json:92]: Pinned to 5.1.x because typescript-eslint otherwise warns about using an unsupported version. Or should we ignore that?

**Comment by johnnyomair** [comment]
Turns out I need this first: https://github.com/vivid-planet/comet/pull/2035#pullrequestreview-2042477964

**Review by dkarnutsch — APPROVED** [review]

**Comment by dkarnutsch** [comment]
Should we consider rolling-up #115 again afterwards?

**Review by nsams — APPROVED** [review]

**Comment by johnnyomair** [comment]
> Should we consider rolling-up #115 again afterwards?

Maybe we should wait until the migration to Next App Router has been finished. One of the open issues relates to typing document types in `src/pages/[[...path]].page.tsx`, which is resolved by using App Router.

**Comment by johnnyomair** [comment]
@dkarnutsch @fraxachun @nsams please review once again. Needed to upgrade react-intl to v6 to resolve a peer dependency error.

**Review by dkarnutsch — APPROVED** [review]

**Review by nsams — APPROVED** [review]

**Pull request merged**